jsonparser.cpp

```diff
@@ -9,23 +9,27 @@ std::map <std::string, std::string> Parser::jsonParser(std::istream& file) {
 }
 
 
+std::string Parser::jsonParserFILE(const std::string& filename){
+	std::string line,alldata;
+	std::ifstream file;
+	file.open(filename);
+	if (file.fail()) throw "The " + filename + " file does not exist or is not readable.";
 
+	while (getline(file, line)) alldata += line;
+	file.close();
+	return alldata;
+}
 
 
 std::map <std::string, std::string> Parser::jsonParser(const std::string& fname) {
 
-	std::string line, alldata, type = "";
+	std::string alldata, type = "";
 	std::map<std::string, std::string> data;
 
 
 	if (fname.find('\"') + 1) alldata = fname;
 	else {
-		std::ifstream file;
-		file.open(fname);
-		if (file.fail()) throw "The " + fname + " file does not exist or is not readable.";
-
-		while (getline(file, line)) alldata += line;
-		file.close();
+		alldata=jsonParserFILE(fname);
```

Comment:
1) Elég zavarba ejtő, hogy a jsonParserFILE egy stringet ad vissza.  Nem erre utal a neve.
2) Az alaphelyzet, hogy jsonParser paramétere lehet egy JSON tartalom, de egy fájl neve is, nem változott meg.